api/index.js

```diff
@@ -1,181 +1,80 @@
-      corrected: record.status.correction !== 'None',
-      courtOrderInPlace: record.status.marginalNote === 'Court order in place',
-      courtOrderRevoked: record.status.marginalNote === 'Court order revoked'
+const requestData = (url, user) => new Promise((resolve, reject) => levRequest.get({
+      'url': url,
+      'headers': user
+        ? { 'X-Auth-Downstream-Username': user }
+        : {}
     },
-    previousRegistration: blocked ? {
-      date: null,
-      systemNumber: null
-    } : {
-      date: record.previousRegistration.date,
-      systemNumber: record.previousRegistration.systemNumber
-    }
-  };
-};
-
-var endpoint = `${config.api.protocol}://${config.api.host}:${config.api.port}/api/v0/events/birth`;
+    oauthUrl,
+    clientId,
+    clientSecret,
+    oAuthUsername,
+    oAuthPassword,
+    helpers.responseHandler(resolve, reject)));
+
+const query = (attrs, user) => {
+  if (attrs === undefined) {
+    throw new ReferenceError('query(): first argument was not defined');
+  } else if (user === undefined) {
+    throw new ReferenceError('query(): second argument was not defined');
+  } else if (!(attrs instanceof Object)) {
+    throw new TypeError('query(): first argument must be an object');
+  } else if (typeof user !== 'string') {
+    throw new TypeError('query(): second argument must be a string');
+  }
 
-var requestData = function requestData(url, user, callback) {
-  return new Promise(function requestDataPromise(resolve, reject) {
-    var headers = user ? { 'X-Auth-Downstream-Username': user } : {};
-    return levRequest.get({
-      'url': url,
-      'headers': headers
-    }, oauthUrl, clientId, clientSecret, oAuthUsername, oAuthPassword,
-      function requestGet(err, res, body) {
-      var statusToName;
-      var statusError;
-      var r;
+  return requestData(helpers.buildQueryUri(endpoint, attrs), user)
+    .then((data) => data.map(helpers.processRecord));
+};
 
-      if (err) {
-        return reject(err);
-      } else if (res.statusCode !== 200) {
-        statusToName = {
-          404: 'NotFoundError',
-          401: 'NotAuthorized'
-        };
+const requestID = (id, user) => {
+  if (id === undefined) {
+    throw new ReferenceError('requestID(): first argument was not defined');
+  } else if (user === undefined) {
+    throw new ReferenceError('requestID(): second argument was not defined');
+  } else if ((!Number.isInteger(id))) {
+    throw new TypeError('requestID(): first argument must be an integer');
+  } else if (typeof user !== 'string') {
+    throw new TypeError('requestID(): second argument must be a string');
+  }
 
-        statusError = new Error(`Received status code "${res.statusCode}" from API`);
+  return requestData(endpoint + '/' + id, user)
+    .then(helpers.processRecord);
+};
 
-        if (statusToName[res.statusCode]) {
-          statusError.name = statusToName[res.statusCode];
-        }
+const read = (attrs, user) => {
```

Comment:
What does read do compared to query? Wonder if the names could be improved for the 3 methods being exported here